src/test/java/nc/students/ayaz/controllers/UserControllerTest.java

```diff
@@ -3,55 +3,58 @@
 import nc.students.ayaz.model.User;
 import nc.students.ayaz.model.Video;
 import nc.students.ayaz.repositories.UserRepository;
-import org.json.JSONArray;
-import org.json.JSONObject;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.test.context.junit4.SpringRunner;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
 
 @RunWith(SpringRunner.class)
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class UserControllerTest {
 
-    @Autowired
+    @MockBean
     private UserRepository repository;
 
     @Autowired
     private TestRestTemplate restTemplate;
 
     @Test
-    public void shouldAddUserToRepositoryWhenPOSTrequestSent() {
+    public void shouldAddUserToRepositoryWhenPostRequestSent() {
         ResponseEntity<String> response = restTemplate.postForEntity("/users/Ayaz", "", String.class);
 
-        assertTrue(repository.contains("Ayaz"));
         assertEquals(HttpStatus.CREATED, response.getStatusCode());
     }
 
     @Test
-    public void shouldReturnUserVideosWhenSendGETrequestWithUsersNickname() throws Exception {
+    public void shouldReturnUserVideosWhenGetRequestSentWithUsersNickname() throws Exception {
         User user = new User("Ayaz");
-        repository.addUser(user);
-        user.addVideo(new Video("FunnyCats"));
-        user.addVideo(new Video("SadDogs"));
-
-        ResponseEntity<String> response = restTemplate.getForEntity("/users/Ayaz", String.class);
-
-        JSONObject jsonFirstVideo = new JSONObject();
-        jsonFirstVideo.put("name", "FunnyCats");
-        JSONObject jsonSecondVideo = new JSONObject();
-        jsonSecondVideo.put("name", "SadDogs");
-        JSONArray jsonVideos = new JSONArray();
-        jsonVideos.put(jsonFirstVideo);
-        jsonVideos.put(jsonSecondVideo);
-        assertEquals(jsonVideos.toString(), response.getBody());
+        Video firstVideo = new Video("FunnyCats");
+        Video secondVideo = new Video("SadDogs");
```

Comment:
I think it's demonstrative. As you said we should test use cases. When user clicks someones nickname then he will see more than 1 video)